airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationHandler.java

```diff
@@ -168,26 +170,32 @@ public DestinationRead getDestination(final DestinationIdRequestBody destination
     return buildDestinationRead(destinationIdRequestBody.getDestinationId());
   }
 
-  public DestinationRead cloneDestination(final DestinationIdRequestBody destinationIdRequestBody)
+  public DestinationRead cloneDestination(final DestinationCloneRequestBody destinationCloneRequestBody)
       throws JsonValidationException, IOException, ConfigNotFoundException {
     // read destination configuration from db
-    final DestinationRead destinationToClone = buildDestinationReadWithSecrets(destinationIdRequestBody.getDestinationId());
-    final ConnectorSpecification spec = getSpec(destinationToClone.getDestinationDefinitionId());
+    final DestinationRead destinationToClone = buildDestinationReadWithSecrets(destinationCloneRequestBody.getDestinationCloneId());
```

Comment:
we should verify that the source / destination being cloned is from the same workspace.